test/adjoint.jl

```diff
@@ -866,3 +866,76 @@ using LinearAlgebra, SciMLSensitivity, OrdinaryDiffEq, ForwardDiff, QuadGK
         end
     end
 end
+
+####Fully oop Adjoint
+
+u0 = @SArray Float32[2.0, 0.0]
+datasize = 30
+tspan = (0.0f0, 1.5f0)
+tsteps = range(tspan[1], tspan[2], length = datasize)
+
```

Comment:
This is a StaticArray/SimpleChain regression test for optimization, not an adjoint test. Please add a proper adjoint test. It should probably be its own file `adjoint_oop.jl` where it similarly tests the direct and AD interfaces for gradient correctness.

This test should be a separate SimpleChains regression test.